design/proof-network.md

```diff
@@ -0,0 +1,257 @@
+    ---------------------              |                ^    ^              |
+    |                   |              |               /      \             |
+    |    provider       |              |              /        \            |
+    |                                                v          v           |
+    |      provider  <--------------------->  validator <----> validator    |
+    |                                                ^          ^           |
+    |  provider         |              |              \        /            |
+    |                   |              |               \      /             |
+    ---------------------              |                v    v              |
+           ^                           |               validator            |
+           |                           |                                    |
+           |                           --------------------------------------
+           |                                          ^
+           |                                          |
+           |             ethereum                     |
+           |                                          |
+           |        ------------------                |
+           |        |                |                |
+           \------- |  marketplace   | <--------------/
+                    |                |
+                    ------------------
+
+Transactions
+------------
+
+The types of transactions that can be included in blocks are:
+
+- `StorageProof(slot id, period, inputs, proof)`
+- `MissingProof(slot id, period, inputs)`
+
+Each transaction is signed by its sender. A *slot id* parameters refers to a
+[slot in a storage request][4] on the marketplace. It uniquely identifies the
+data for which a storage proof is required. The *period* refers to a [time
+interval][5] in which the storage proof should be submitted. By *proof* we mean
+a [zero-knowledge proof][5], and by *inputs* we mean its public inputs.
+
+#### StorageProof ####
+
+A storage provider sends a `StorageProof` transaction to a validator to indicate
+to the network that it calculated a storage proof as required by the
+marketplace. This validator includes it in its next block.
+
+#### MissingProof ####
+
+A validator includes a `MissingProof` transaction in its next block when it
+notices that a required storage proof was not submitted.
+
+[4]: marketplace.md
+[5]: https://github.com/codex-storage/codex-storage-proofs-circuits#circuit
+
+Flows
+-------
+
+### Successfull proof submission ###
+
+Storage providers monitor the on-chain marketplace to check in which periods
+they need to provide a storage proof. When a provider sees that a proof is
+required for a slot in the current *period*, it gathers public *inputs* for the
+slot, including the random challenge and calculates a zero-knowledge storage
+*proof*. The provider then submits `StorageProof(slot id, period, inputs,
+proof)` to a single validator.
```

Comment:
Why a single validator? What if this one is malicious?

Reply:
Each validator produces a block every round in the Mysticeti design, and the proof only needs to end up in one of them. If the validator is malicious, it won't be able to give an inclusion proof for the transaction, and you can simply go to another validator to get the transaction included.